bundle/libraries/same_name_libraries.go

```diff
@@ -0,0 +1,101 @@
+package libraries
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type checkForSameNameLibraries struct{}
+
+var patterns = []dyn.Pattern{
+	taskLibrariesPattern.Append(dyn.AnyIndex(), dyn.AnyKey()),
+	forEachTaskLibrariesPattern.Append(dyn.AnyIndex(), dyn.AnyKey()),
+	envDepsPattern.Append(dyn.AnyIndex()),
+}
+
+type libData struct {
+	fullPath  string
+	locations []dyn.Location
+	paths     []dyn.Path
+}
+
+func (c checkForSameNameLibraries) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	var diags diag.Diagnostics
+	libs := make(map[string]*libData)
+
+	err := b.Config.Mutate(func(v dyn.Value) (dyn.Value, error) {
+		var err error
+		for _, pattern := range patterns {
+			v, err = dyn.MapByPattern(v, pattern, func(p dyn.Path, lv dyn.Value) (dyn.Value, error) {
+				libPath, ok := lv.AsString()
+				if !ok {
+					return lv, nil
+				}
+
+				// If not local library, skip the check
+				if !IsLibraryLocal(libPath) {
+					return lv, nil
+				}
+
+				libFullPath := lv.MustString()
```

Comment:
Why are you calling `lv.MustString()` here, it seems you already unpacked lv above?

Reply:
Good catch, should exisiting value from above